### PR TITLE
Add streaming ST/SLU

### PR DESCRIPTION
**Pull request by D-Keqi**
This PR is to add streaming ST/SLU.

### CODE REVIEW

**Comment by D-Keqi** [comment]
> The PR looks good to me. It would be good to add some related test functions, but we can do it later in some following PRs

Hi, do you mean integration test https://github.com/espnet/espnet/blob/master/ci/test_integration_espnet2.sh or something like test_st_inference_streaming.py https://github.com/espnet/espnet/tree/master/test/espnet2/bin ?

**Comment by D-Keqi** [comment]
> * yes, adding one example to https://github.com/espnet/espnet/blob/master/ci/test_integration_espnet2.sh is fine
> * please also merge the latest master branch.

OK!

**Comment by D-Keqi** [comment]
@ftshijt Hi! Could I have a discussion with you about the integration test https://github.com/espnet/espnet/blob/master/ci/test_integration_espnet2.sh ? In the integration test https://github.com/espnet/espnet/blob/master/ci/test_integration_espnet2.sh, we always use the 'mini_an4' example to have a test. But it seems that this is not feasible for ST, if we use fisher_callhome, it is too big for a test I think. What do you think about this?

**Comment by ftshijt** [comment]
> @ftshijt Hi! Could I have a discussion with you about the integration test https://github.com/espnet/espnet/blob/master/ci/test_integration_espnet2.sh ? In the integration test https://github.com/espnet/espnet/blob/master/ci/test_integration_espnet2.sh, we always use the 'mini_an4' example to have a test. But it seems that this is not feasible for ST, if we use fisher_callhome, it is too big for a test I think. What do you think about this?

To keep it aligned, it might be good that we just add some dummy outputs for st over `mini_an4` so that we can still use the same testing. In general, we won't care about the performance, but just to make sure it is stably run.

**Comment by ftshijt** [comment]
Another possibility is to hold a good translation elsewhere for testing only.

**Comment by D-Keqi** [comment]
> Another possibility is to hold a good translation elsewhere for testing only.

OK, so can we merge this PR first? then I can make a demo using the latest ESPnet and I think this demo can also be viewed as a test.

**Comment by sw005320** [comment]
@D-Keqi, a demo is a good test, but the integration test would be safer to make your implementation solid.
So, can you just add a test as @ftshijt suggested?

**Comment by D-Keqi** [comment]
> @D-Keqi, a demo is a good test, but the integration test would be safer to make your implementation solid. So, can you just add a test as @ftshijt suggested?

Sure!

**Comment by D-Keqi** [comment]
@ftshijt Hi, Could I have some discussions with you? It seems that there is an issue when using fbank_pitch for ST ESPnet2, when you write the st.sh, have you tested this condition or come into the same issue?
The error issue is reported as follows:
'./run.sh --stage 2 --stop-stage 4 --feats-type fbank_pitch'
2022-04-20T03:11:02 (st.sh:261:main) ./st.sh --src_lang en --tgt_lang en --src_token_type bpe --src_nbpe 30 --tgt_token_type bpe --tgt_nbpe 30 --src_case lc.rm --tgt_case lc.rm --src_bpe_train_text data/train_nodev/text.lc.rm.en --tgt_bpe_train_text data/train_nodev/text.lc.rm.en --use_lm false --token_joint false --st_config conf/train_st.yaml --train_set train_nodev --valid_set train_dev --test_sets train_dev test test_seg --lm_train_text data/train_nodev/text.lc.rm.en --stage 2 --stop-stage 4 --feats-type fbank_pitch
2022-04-20T03:11:02 (st.sh:521:main) Skip stage 2: Speed perturbation
2022-04-20T03:11:02 (st.sh:584:main) [Require Kaldi] Stage 3: fbank_pitch extract: data/ -> dump/fbank_pitch
utils/copy_data_dir.sh: copied data from data/train_nodev to dump/fbank_pitch/org/train_nodev
utils/validate_data_dir.sh: Successfully validated data-directory dump/fbank_pitch/org/train_nodev
steps/make_fbank_pitch.sh --nj 4 --cmd run.pl dump/fbank_pitch/org/train_nodev
utils/validate_data_dir.sh: Successfully validated data-directory dump/fbank_pitch/org/train_nodev
steps/make_fbank_pitch.sh: [info]: no segments file exists: assuming wav.scp indexed by utterance.
run.pl: 4 / 4 failed, log is in dump/fbank_pitch/org/train_nodev/log/make_fbank_pitch_train_nodev.*.log

And the error report in these log are as follows:
paste-feats --length-tolerance=2 'ark:compute-fbank-feats  --write-utt2dur=ark,t:dump/fbank_pitch/org/train_nodev/log/utt2dur.1    --verbose=2 --config=conf/fbank.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:- |' 'ark,s,cs:compute-kaldi-pitch-feats --verbose=2       --config=conf/pitch.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:- |     process-kaldi-pitch-feats  ark:- ark:- |' ark:-
copy-feats --compress=true --write-num-frames=ark,t:dump/fbank_pitch/org/train_nodev/log/utt2num_frames.1 ark:- ark,scp:/home/dengkeqi/tools/ST/newone/espnet/egs2/mini_an4/st1/dump/fbank_pitch/org/train_nodev/data/raw_fbank_pitch_train_nodev.1.ark,/home/dengkeqi/tools/ST/newone/espnet/egs2/mini_an4/st1/dump/fbank_pitch/org/train_nodev/data/raw_fbank_pitch_train_nodev.1.scp
compute-fbank-feats --write-utt2dur=ark,t:dump/fbank_pitch/org/train_nodev/log/utt2dur.1 --verbose=2 --config=conf/fbank.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:-
ERROR (compute-fbank-feats[5.5.973~1-48ad]:ComputeFeatures():feat/feature-common-inl.h:41) Waveform and config sample Frequency mismatch: 16000 .vs 8000 (use --allow-downsample=true to allow  downsampling the waveform).

[ Stack-Trace: ]
/root/toolkits/kaldi/src/lib/libkaldi-base.so(kaldi::MessageLogger::LogMessage() const+0xb42) [0x7f38a26ca732]
compute-fbank-feats(kaldi::MessageLogger::LogAndThrow::operator=(kaldi::MessageLogger const&)+0x21) [0x55c6bc1bf2a5]
compute-fbank-feats(kaldi::OfflineFeatureTpl<kaldi::FbankComputer>::ComputeFeatures(kaldi::VectorBase<float> const&, float, float, kaldi::Matrix<float>*)+0x1a3) [0x55c6bc1c68d7]
compute-fbank-feats(main+0xbd1) [0x55c6bc1bcaab]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7f38a1b34bf7]
compute-fbank-feats(_start+0x2a) [0x55c6bc1bbdfa]

WARNING (compute-fbank-feats[5.5.973~1-48ad]:main():compute-fbank-feats.cc:151) Failed to compute features for utterance fash-an253-b
LOG (compute-fbank-feats[5.5.973~1-48ad]:main():compute-fbank-feats.cc:185)  Done 0 out of 1 utterances.
process-kaldi-pitch-feats ark:- ark:-
compute-kaldi-pitch-feats --verbose=2 --config=conf/pitch.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:-
ERROR (compute-kaldi-pitch-feats[5.5.973~1-48ad]:main():compute-kaldi-pitch-feats.cc:88) Sample frequency mismatch: you specified 8000 but data has 16000 (use --sample-frequency option).  Utterance is fash-an253-b

[ Stack-Trace: ]
/root/toolkits/kaldi/src/lib/libkaldi-base.so(kaldi::MessageLogger::LogMessage() const+0xb42) [0x7fd51e2c2732]
compute-kaldi-pitch-feats(kaldi::MessageLogger::LogAndThrow::operator=(kaldi::MessageLogger const&)+0x21) [0x55c277815923]
compute-kaldi-pitch-feats(main+0x4b5) [0x55c2778148df]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7fd51d72cbf7]
compute-kaldi-pitch-feats(_start+0x2a) [0x55c27781434a]

kaldi::KaldiFatalErrorLOG (process-kaldi-pitch-feats[5.5.973~1-48ad]:main():process-kaldi-pitch-feats.cc:85) Post-processed pitch for 0 utterances.
WARNING (paste-feats[5.5.973~1-48ad]:Close():kaldi-io.cc:515) Pipe compute-kaldi-pitch-feats --verbose=2       --config=conf/pitch.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:- |     process-kaldi-pitch-feats  ark:- ark:- | had nonzero return status 256
LOG (paste-feats[5.5.973~1-48ad]:main():paste-feats.cc:158) Done 0 utts, errors on 0
WARNING (paste-feats[5.5.973~1-48ad]:Close():kaldi-io.cc:515) Pipe compute-fbank-feats  --write-utt2dur=ark,t:dump/fbank_pitch/org/train_nodev/log/utt2dur.1    --verbose=2 --config=conf/fbank.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:- | had nonzero return status 256
ERROR (paste-feats[5.5.973~1-48ad]:~SequentialTableReaderArchiveImpl():util/kaldi-table-inl.h:678) TableReader: error detected closing archive 'compute-fbank-feats  --write-utt2dur=ark,t:dump/fbank_pitch/org/train_nodev/log/utt2dur.1    --verbose=2 --config=conf/fbank.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:- |'

[ Stack-Trace: ]
/root/toolkits/kaldi/src/lib/libkaldi-base.so(kaldi::MessageLogger::LogMessage() const+0xb42) [0x7fab5ad05732]
paste-feats(kaldi::MessageLogger::LogAndThrow::operator=(kaldi::MessageLogger const&)+0x21) [0x5580dec3f97b]
paste-feats(kaldi::SequentialTableReaderArchiveImpl<kaldi::KaldiObjectHolder<kaldi::Matrix<float> > >::~SequentialTableReaderArchiveImpl()+0x18c) [0x5580dec41d10]
paste-feats(kaldi::SequentialTableReaderArchiveImpl<kaldi::KaldiObjectHolder<kaldi::Matrix<float> > >::~SequentialTableReaderArchiveImpl()+0x9) [0x5580dec41d1f]
paste-feats(kaldi::SequentialTableReader<kaldi::KaldiObjectHolder<kaldi::Matrix<float> > >::~SequentialTableReader()+0x12) [0x5580dec425a2]
paste-feats(main+0x8f1) [0x5580dec3e704]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7fab5a16fbf7]
paste-feats(_start+0x2a) [0x5580dec3d92a]

terminate called after throwing an instance of 'kaldi::KaldiFatalError'
  what():  kaldi::KaldiFatalError
LOG (copy-feats[5.5.973~1-48ad]:main():copy-feats.cc:143) Copied 0 feature matrices.
# Accounting: time=0 threads=1

**Comment by D-Keqi** [comment]
@ftshijt Hi, Could I have some discussions with you? It seems that there is an issue when using fbank_pitch for ST ESPnet2, but it works for raw feature. When you write the st.sh, have you tested this condition or come into the same issue?
The error issue is reported as follows:
'./run.sh --stage 2 --stop-stage 4 --feats-type fbank_pitch'
2022-04-20T03:11:02 (st.sh:261:main) ./st.sh --src_lang en --tgt_lang en --src_token_type bpe --src_nbpe 30 --tgt_token_type bpe --tgt_nbpe 30 --src_case lc.rm --tgt_case lc.rm --src_bpe_train_text data/train_nodev/text.lc.rm.en --tgt_bpe_train_text data/train_nodev/text.lc.rm.en --use_lm false --token_joint false --st_config conf/train_st.yaml --train_set train_nodev --valid_set train_dev --test_sets train_dev test test_seg --lm_train_text data/train_nodev/text.lc.rm.en --stage 2 --stop-stage 4 --feats-type fbank_pitch
2022-04-20T03:11:02 (st.sh:521:main) Skip stage 2: Speed perturbation
2022-04-20T03:11:02 (st.sh:584:main) [Require Kaldi] Stage 3: fbank_pitch extract: data/ -> dump/fbank_pitch
utils/copy_data_dir.sh: copied data from data/train_nodev to dump/fbank_pitch/org/train_nodev
utils/validate_data_dir.sh: Successfully validated data-directory dump/fbank_pitch/org/train_nodev
steps/make_fbank_pitch.sh --nj 4 --cmd run.pl dump/fbank_pitch/org/train_nodev
utils/validate_data_dir.sh: Successfully validated data-directory dump/fbank_pitch/org/train_nodev
steps/make_fbank_pitch.sh: [info]: no segments file exists: assuming wav.scp indexed by utterance.
run.pl: 4 / 4 failed, log is in dump/fbank_pitch/org/train_nodev/log/make_fbank_pitch_train_nodev.*.log

And the error report in these log are as follows:
paste-feats --length-tolerance=2 'ark:compute-fbank-feats  --write-utt2dur=ark,t:dump/fbank_pitch/org/train_nodev/log/utt2dur.1    --verbose=2 --config=conf/fbank.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:- |' 'ark,s,cs:compute-kaldi-pitch-feats --verbose=2       --config=conf/pitch.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:- |     process-kaldi-pitch-feats  ark:- ark:- |' ark:-
copy-feats --compress=true --write-num-frames=ark,t:dump/fbank_pitch/org/train_nodev/log/utt2num_frames.1 ark:- ark,scp:/home/dengkeqi/tools/ST/newone/espnet/egs2/mini_an4/st1/dump/fbank_pitch/org/train_nodev/data/raw_fbank_pitch_train_nodev.1.ark,/home/dengkeqi/tools/ST/newone/espnet/egs2/mini_an4/st1/dump/fbank_pitch/org/train_nodev/data/raw_fbank_pitch_train_nodev.1.scp
compute-fbank-feats --write-utt2dur=ark,t:dump/fbank_pitch/org/train_nodev/log/utt2dur.1 --verbose=2 --config=conf/fbank.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:-
ERROR (compute-fbank-feats[5.5.973~1-48ad]:ComputeFeatures():feat/feature-common-inl.h:41) Waveform and config sample Frequency mismatch: 16000 .vs 8000 (use --allow-downsample=true to allow  downsampling the waveform).

[ Stack-Trace: ]
/root/toolkits/kaldi/src/lib/libkaldi-base.so(kaldi::MessageLogger::LogMessage() const+0xb42) [0x7f38a26ca732]
compute-fbank-feats(kaldi::MessageLogger::LogAndThrow::operator=(kaldi::MessageLogger const&)+0x21) [0x55c6bc1bf2a5]
compute-fbank-feats(kaldi::OfflineFeatureTpl<kaldi::FbankComputer>::ComputeFeatures(kaldi::VectorBase<float> const&, float, float, kaldi::Matrix<float>*)+0x1a3) [0x55c6bc1c68d7]
compute-fbank-feats(main+0xbd1) [0x55c6bc1bcaab]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7f38a1b34bf7]
compute-fbank-feats(_start+0x2a) [0x55c6bc1bbdfa]

WARNING (compute-fbank-feats[5.5.973~1-48ad]:main():compute-fbank-feats.cc:151) Failed to compute features for utterance fash-an253-b
LOG (compute-fbank-feats[5.5.973~1-48ad]:main():compute-fbank-feats.cc:185)  Done 0 out of 1 utterances.
process-kaldi-pitch-feats ark:- ark:-
compute-kaldi-pitch-feats --verbose=2 --config=conf/pitch.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:-
ERROR (compute-kaldi-pitch-feats[5.5.973~1-48ad]:main():compute-kaldi-pitch-feats.cc:88) Sample frequency mismatch: you specified 8000 but data has 16000 (use --sample-frequency option).  Utterance is fash-an253-b

[ Stack-Trace: ]
/root/toolkits/kaldi/src/lib/libkaldi-base.so(kaldi::MessageLogger::LogMessage() const+0xb42) [0x7fd51e2c2732]
compute-kaldi-pitch-feats(kaldi::MessageLogger::LogAndThrow::operator=(kaldi::MessageLogger const&)+0x21) [0x55c277815923]
compute-kaldi-pitch-feats(main+0x4b5) [0x55c2778148df]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7fd51d72cbf7]
compute-kaldi-pitch-feats(_start+0x2a) [0x55c27781434a]

kaldi::KaldiFatalErrorLOG (process-kaldi-pitch-feats[5.5.973~1-48ad]:main():process-kaldi-pitch-feats.cc:85) Post-processed pitch for 0 utterances.
WARNING (paste-feats[5.5.973~1-48ad]:Close():kaldi-io.cc:515) Pipe compute-kaldi-pitch-feats --verbose=2       --config=conf/pitch.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:- |     process-kaldi-pitch-feats  ark:- ark:- | had nonzero return status 256
LOG (paste-feats[5.5.973~1-48ad]:main():paste-feats.cc:158) Done 0 utts, errors on 0
WARNING (paste-feats[5.5.973~1-48ad]:Close():kaldi-io.cc:515) Pipe compute-fbank-feats  --write-utt2dur=ark,t:dump/fbank_pitch/org/train_nodev/log/utt2dur.1    --verbose=2 --config=conf/fbank.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:- | had nonzero return status 256
ERROR (paste-feats[5.5.973~1-48ad]:~SequentialTableReaderArchiveImpl():util/kaldi-table-inl.h:678) TableReader: error detected closing archive 'compute-fbank-feats  --write-utt2dur=ark,t:dump/fbank_pitch/org/train_nodev/log/utt2dur.1    --verbose=2 --config=conf/fbank.conf scp,p:dump/fbank_pitch/org/train_nodev/log/wav_train_nodev.1.scp ark:- |'

[ Stack-Trace: ]
/root/toolkits/kaldi/src/lib/libkaldi-base.so(kaldi::MessageLogger::LogMessage() const+0xb42) [0x7fab5ad05732]
paste-feats(kaldi::MessageLogger::LogAndThrow::operator=(kaldi::MessageLogger const&)+0x21) [0x5580dec3f97b]
paste-feats(kaldi::SequentialTableReaderArchiveImpl<kaldi::KaldiObjectHolder<kaldi::Matrix<float> > >::~SequentialTableReaderArchiveImpl()+0x18c) [0x5580dec41d10]
paste-feats(kaldi::SequentialTableReaderArchiveImpl<kaldi::KaldiObjectHolder<kaldi::Matrix<float> > >::~SequentialTableReaderArchiveImpl()+0x9) [0x5580dec41d1f]
paste-feats(kaldi::SequentialTableReader<kaldi::KaldiObjectHolder<kaldi::Matrix<float> > >::~SequentialTableReader()+0x12) [0x5580dec425a2]
paste-feats(main+0x8f1) [0x5580dec3e704]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7fab5a16fbf7]
paste-feats(_start+0x2a) [0x5580dec3d92a]

terminate called after throwing an instance of 'kaldi::KaldiFatalError'
  what():  kaldi::KaldiFatalError
LOG (copy-feats[5.5.973~1-48ad]:main():copy-feats.cc:143) Copied 0 feature matrices.
# Accounting: time=0 threads=1

**Comment by D-Keqi** [comment]
@ftshijt @sw005320 Could I ask for some advice or help? Currently, I still have a CI issue related with offline ST decoding when using integration test. But the issue report is in the "exp/st_train_st_raw_bpe_lc.rm30_max_epoch1/inference_beam_size1_st_model_valid.acc.ave/train_dev/logdir/st_inference.1.log", is it possible that I can see the content of the log file in the Web UI? 
I have make a local intergration test using the latest ESPnet, it works without any issue reported. This is weird, but I currently do not have an idea of how to solve this.

**Comment by sw005320** [comment]
> @ftshijt @sw005320 Could I ask for some advice or help? Currently, I still have a CI issue related with offline ST decoding when using integration test. But the issue report is in the "exp/st_train_st_raw_bpe_lc.rm30_max_epoch1/inference_beam_size1_st_model_valid.acc.ave/train_dev/logdir/st_inference.1.log", is it possible that I can see the content of the log file in the Web UI? I have made a local intergration test using the latest ESPnet, it works without any issue reported. This is weird, but I currently do not have an idea of how to fix this CI issue.

I'll check it, but is it specific to some PyTorch versions in CI?

**Comment by D-Keqi** [comment]
> > @ftshijt @sw005320 Could I ask for some advice or help? Currently, I still have a CI issue related with offline ST decoding when using integration test. But the issue report is in the "exp/st_train_st_raw_bpe_lc.rm30_max_epoch1/inference_beam_size1_st_model_valid.acc.ave/train_dev/logdir/st_inference.1.log", is it possible that I can see the content of the log file in the Web UI? I have made a local intergration test using the latest ESPnet, it works without any issue reported. This is weird, but I currently do not have an idea of how to fix this CI issue.
> 
> I'll check it, but is it specific to some PyTorch versions in CI?

Mnay thanks!
I do not have too much knowledge about this, I think it might be possible, and the Pytorch version of my local test is 1.7.1

**Comment by sw005320** [comment]
Sorry, @D-Keqi, I could not find a way to do it.
Maybe, we could hack `asr.sh` in this PR to display the error for this particular issue...

One more comment.
I recommend you add a unit test for st inference, similar to https://github.com/espnet/espnet/blob/master/test/espnet2/bin/test_asr_inference.py
However, I found that we do not have such for st.

@ftshijt,
can I ask you to make a unit test for st?
I think we need to have `test_st_inference.py` and `test_st_train.py`.
We can just follow the ASR ones.
We may also need to have that for mt as well.

**Comment by D-Keqi** [comment]
> Sorry, @D-Keqi, I could not find a way to do it. Maybe, we could hack `asr.sh` in this PR to display the error for this particular issue...
> 
> One more comment. I recommend you add a unit test for st inference, similar to https://github.com/espnet/espnet/blob/master/test/espnet2/bin/test_asr_inference.py However, I found that we do not have such for st.
> 
> @ftshijt, can I ask you to make a unit test for st? I think we need to have `test_st_inference.py` and `test_st_train.py`. We can just follow the ASR ones. We may also need to have that for mt as well.

It seems that I have successfully fixed this issue and passed the integration test. Is the unit test still in need? @sw005320

**Comment by sw005320** [comment]
> It seems that I have successfully fixed this issue and passed the integration test. Is the unit test still in need? @sw005320

Very cool.
Yes, we need this, but it should be first triggered by @ftshijt.
@ftshijt, can you make the above unit tests?
Then, I want @D-Keqi to add the ST streaming related unit test.

File: espnet2/bin/st_inference.py [hunk 1]
```diff
@@ -306,7 +306,7 @@ def inference(
     token_type: Optional[str],
     bpemodel: Optional[str],
     allow_variable_data_keys: bool,
-    enh_s2t_task: bool,
+    enh_s2t_task: bool = False,
```
Comment (sw005320) [inline espnet2/bin/st_inference.py:309]: @simpleoier, is it OK?

**Comment by ftshijt** [comment]
> > It seems that I have successfully fixed this issue and passed the integration test. Is the unit test still in need? @sw005320
> 
> Very cool. Yes, we need this, but it should be first triggered by @ftshijt. @ftshijt, can you make the above unit tests? Then, I want @D-Keqi to add the ST streaming related unit test.

Sure, I will do it after tomorrow's meetings

File: espnet2/bin/st_inference.py (hunk 1 above)
Comment (sw005320) [reply espnet2/bin/st_inference.py:309]: @D-Keqi, can you tell me why you specify it `False`?

**Comment by sw005320** [comment]
@D-Keqi, after we fix this conflict, and clarify my question about `enh_s2t_task: bool = False`, I'll merge this PR.

File: espnet2/bin/st_inference.py (hunk 1 above)
Comment (simpleoier) [reply espnet2/bin/st_inference.py:309]: Sorry @sw005320, it's my fault. I missed one argument definition in the pull request. I put it in the [asr_inference](https://github.com/espnet/espnet/blob/40202fca64529982fa7c1c428137243f306a0d5d/espnet2/bin/asr_inference.py#L553), but not in the st_inference. @D-Keqi Can you wait for my fix or fix it in this PR?

Comment (simpleoier) [reply espnet2/bin/st_inference.py:309]: I submitted a PR #4323. If @D-Keqi fix it in this PR, I can close the other one.

**Comment by D-Keqi** [comment]
> @D-Keqi, after we fix this conflict, and clarify my question about `enh_s2t_task: bool = False`, I'll merge this PR.

OK!

**Comment by D-Keqi** [comment]
It is a little weird why this conflict is still not fixed. Can I get some suggestions? @sw005320

**Comment by ftshijt** [comment]
FYI: I'm adding st_inference and st_train in tests https://github.com/espnet/espnet/pull/4324

File: espnet2/bin/st_inference.py (hunk 1 above)
Comment (sw005320) [inline espnet2/bin/st_inference.py:309]:
```suggestion
    enh_s2t_task: bool,
```
We do not need this change with #4323

Comment (D-Keqi) [reply espnet2/bin/st_inference.py:309]: OK, but acutally it seems can not pass the integration offline ST test with "enh_s2t_task: bool", therefore I add "enh_s2t_task: bool = False".

Comment (D-Keqi) [reply espnet2/bin/st_inference.py:309]: @simpleoier thanks! I think based on your PR, my modification to the st_inference.py is not needed.

**Comment by D-Keqi** [comment]
I think this PR is OK to be merged. @sw005320

**Comment by sw005320** [comment]
super thanks, @D-Keqi!

**Comment by chenzhehuai** [comment]
Hi, this mail address is unused now. The mail will be forwarded to ***@***.******@***.***

**Pull request merged**